ci/flake: TestUsers_PasswordResetExpiry disabled

### Test

TestUsers_PasswordResetExpiry/expired_link

### Example failure

https://buildkite.com/sourcegraph/sourcegraph/builds/193660#0185bb03-890a-486f-a119-e5e80dd2c29e

### Disabling PR

https://github.com/sourcegraph/sourcegraph/pull/46508

### Additional details

_No response_

/cc @sourcegraph/source/iam

Copying from the build failures, this seems to be a data race:

```


=== FAIL: internal/database TestUsers_PasswordResetExpiry/expired_link (0.02s)
--
  | ==================
  | WARNING: DATA RACE
  | Write at 0x00c00044c2f0 by goroutine 25717:
  | github.com/sourcegraph/sourcegraph/internal/conf.(*store).Mock()
  | /root/buildkite/build/sourcegraph/internal/conf/store.go:79 +0x84
  | github.com/sourcegraph/sourcegraph/internal/conf.(*client).Mock()
  | /root/buildkite/build/sourcegraph/internal/conf/client.go:123 +0x47
  | github.com/sourcegraph/sourcegraph/internal/conf.Mock()
  | /root/buildkite/build/sourcegraph/internal/conf/client.go:118 +0x25
  | github.com/sourcegraph/sourcegraph/internal/database.TestUsers_PasswordResetExpiry.func1()
  | /root/buildkite/build/sourcegraph/internal/database/users_builtin_auth_test.go:326 +0x104
  | testing.tRunner()
  | /root/.asdf/installs/golang/1.19.3/go/src/testing/testing.go:1446 +0x216
  | testing.(*T).Run.func1()
  | /root/.asdf/installs/golang/1.19.3/go/src/testing/testing.go:1493 +0x47
  |  
  | Previous read at 0x00c00044c2f0 by goroutine 3934:
  | github.com/sourcegraph/sourcegraph/internal/conf.(*store).Raw()
  | /root/buildkite/build/sourcegraph/internal/conf/store.go:60 +0x105
  | github.com/sourcegraph/sourcegraph/internal/conf.(*client).Raw()
  | /root/buildkite/build/sourcegraph/internal/conf/client.go:86 +0x84
  | github.com/sourcegraph/sourcegraph/internal/conf.Raw()
  | /root/buildkite/build/sourcegraph/internal/conf/client.go:54 +0x60
  | github.com/sourcegraph/sourcegraph/internal/conf.ValidateSite()
  | /root/buildkite/build/sourcegraph/internal/conf/validate.go:188 +0x4a
  | github.com/sourcegraph/sourcegraph/internal/database.(*confStore).createIfUpToDate()
  | /root/buildkite/build/sourcegraph/internal/database/conf.go:222 +0x164
  | github.com/sourcegraph/sourcegraph/internal/database.(*confStore).SiteCreateIfUpToDate()
  | /root/buildkite/build/sourcegraph/internal/database/conf.go:118 +0x252
  | github.com/sourcegraph/sourcegraph/internal/database.TestSiteCreateIfUpToDate.func1()
  | /root/buildkite/build/sourcegraph/internal/database/conf_test.go:199 +0x28b
  | testing.tRunner()
  | /root/.asdf/installs/golang/1.19.3/go/src/testing/testing.go:1446 +0x216
  | testing.(*T).Run.func1()
  | /root/.asdf/installs/golang/1.19.3/go/src/testing/testing.go:1493 +0x47
  |  
  | Goroutine 25717 (running) created at:
  | testing.(*T).Run()
  | /root/.asdf/installs/golang/1.19.3/go/src/testing/testing.go:1493 +0x75d
  | github.com/sourcegraph/sourcegraph/internal/database.TestUsers_PasswordResetExpiry()
  | /root/buildkite/build/sourcegraph/internal/database/users_builtin_auth_test.go:325 +0x5b5
  | testing.tRunner()
  | /root/.asdf/installs/golang/1.19.3/go/src/testing/testing.go:1446 +0x216
  | testing.(*T).Run.func1()
  | /root/.asdf/installs/golang/1.19.3/go/src/testing/testing.go:1493 +0x47
  |  
  | Goroutine 3934 (finished) created at:
  | testing.(*T).Run()
  | /root/.asdf/installs/golang/1.19.3/go/src/testing/testing.go:1493 +0x75d
  | github.com/sourcegraph/sourcegraph/internal/database.TestSiteCreateIfUpToDate()
  | /root/buildkite/build/sourcegraph/internal/database/conf_test.go:194 +0x9e9
  | testing.tRunner()
  | /root/.asdf/installs/golang/1.19.3/go/src/testing/testing.go:1446 +0x216
  | testing.(*T).Run.func1()
  | /root/.asdf/installs/golang/1.19.3/go/src/testing/testing.go:1493 +0x47
  | ==================
  | testing.go:1319: race detected during execution of test
  | --- FAIL: TestUsers_PasswordResetExpiry/expired_link (0.02s)


```

--- a/internal/database/users_builtin_auth_test.go
+++ b/internal/database/users_builtin_auth_test.go
@@ -3,8 +3,8 @@ package database
 import (
 	"context"
 	"testing"
-	"time"
 
+	"github.com/keegancsmith/sqlf"
 	"github.com/sourcegraph/log/logtest"
 
 	"github.com/sourcegraph/sourcegraph/internal/conf"
@@ -301,12 +301,21 @@ func TestUsers_PasswordResetExpiry(t *testing.T) {
 	if testing.Short() {
 		t.Skip()
 	}
-	t.Parallel()
+
 	logger := logtest.Scoped(t)
 	db := NewDB(logger, dbtest.NewDB(logger, t))
 	ctx := context.Background()
 
-	user, err := db.Users().Create(ctx, NewUser{
+	// We setup the configuration so that password reset links are valid for 60s
+	conf.Mock(&conf.Unified{
+		SiteConfiguration: schema.SiteConfiguration{
+			AuthPasswordResetLinkExpiry: 60,
+		},
+	})
+	defer conf.Mock(nil)
+
+	users := db.Users()
+	user, err := users.Create(ctx, NewUser{
 		Email:                 "foo@bar.com",
 		Username:              "foo",
 		Password:              "right-password",
@@ -316,45 +325,37 @@ func TestUsers_PasswordResetExpiry(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	resetCode, err := db.Users().RenewPasswordResetCode(ctx, user.ID)
+	resetCode, err := users.RenewPasswordResetCode(ctx, user.ID)
 	if err != nil {
 		t.Fatal(err)
 	}
-	time.Sleep(time.Second) // the lowest expiry is 1 second
 
-	t.Run("expired link", func(t *testing.T) {
-		// This flaked with a data race in https://buildkite.com/sourcegraph/sourcegraph/builds/193660#0185bb03-890a-486f-a119-e5e80dd2c29e
-		t.Skip()
-		conf.Mock(&conf.Unified{
-			SiteConfiguration: schema.SiteConfiguration{
-				AuthPasswordResetLinkExpiry: 1,
-			},
-		})
-		defer conf.Mock(nil)
+	// Reset the passwd_reset_time to be 5min in the past, so that it's expired
+	_, err = users.ExecResult(ctx, sqlf.Sprintf("UPDATE users SET passwd_reset_time = now()-'5 minutes'::interval WHERE users.id = %s", user.ID))
+	if err != nil {
+		t.Fatalf("failed to update reset time: %s", err)
+	}
 
-		success, err := db.Users().SetPassword(ctx, user.ID, resetCode, "new-password")
-		if err != nil {
-			t.Fatal(err)
-		}
-		if success {
-			t.Fatal("accepted an expired password reset")
-		}
-	})
+	// This should fail, because it has been reset 5min ago, but link is only valid 60s
+	success, err := users.SetPassword(ctx, user.ID, resetCode, "new-password")
+	if err != nil {
+		t.Fatal(err)
+	}
+	if success {
+		t.Fatal("accepted an expired password reset")
+	}
 
-	t.Run("valid link", func(t *testing.T) {
-		conf.Mock(&conf.Unified{
-			SiteConfiguration: schema.SiteConfiguration{
-				AuthPasswordResetLinkExpiry: 3600,
-			},
-		})
-		defer conf.Mock(nil)
+	// Now we want the link to be fresh by setting passwd_reset_time to now
+	_, err = users.ExecResult(ctx, sqlf.Sprintf("UPDATE users SET passwd_reset_time = now() WHERE users.id = %s", user.ID))
+	if err != nil {
+		t.Fatalf("failed to update reset time: %s", err)
+	}
 
-		success, err := db.Users().SetPassword(ctx, user.ID, resetCode, "new-password")
-		if err != nil {
-			t.Fatal(err)
-		}
-		if !success {
-			t.Fatal("did not accept a valid password reset")
-		}
-	})
+	success, err = users.SetPassword(ctx, user.ID, resetCode, "new-password")
+	if err != nil {
+		t.Fatal(err)
+	}
+	if !success {
+		t.Fatal("did not accept a valid password reset")
+	}
 }

--- a/internal/database/users_builtin_auth_test.go
+++ b/internal/database/users_builtin_auth_test.go
@@ -312,7 +312,7 @@ func TestUsers_PasswordResetExpiry(t *testing.T) {
 			AuthPasswordResetLinkExpiry: 60,
 		},
 	})
-	defer conf.Mock(nil)
+	t.Cleanup(func() { conf.Mock(nil)})
 
 	users := db.Users()
 	user, err := users.Create(ctx, NewUser{

